message/retrylater.go

```diff
@@ -23,19 +23,17 @@ type retryLaterHandler struct {
 }
 
 func (r *retryLaterHandler) Do(ctx context.Context, orig Handler, message *servicebus.Message) Handler {
-	go func() {
-		ctx, span := tracing.StartSpanFromMessageAndContext(ctx, "go-shuttle.retryLater.Do", message)
-		defer span.End()
+	ctx, span := tracing.StartSpanFromMessageAndContext(ctx, "go-shuttle.retryLater.Do", message)
+	defer span.End()
 
-		select {
-		// TODO this can go past lock duration pretty easily
-		// Ideally we'd also timeout at dequeue time + lockdurtaion - 1 second but don't have access to dequeue time
-		// Maybe we can use context.WithTimeout when we recieve the message?
-		case <-ctx.Done():
-			span.Logger().Error(errors.Wrap(ctx.Err(), "Retry context expired"))
-		case <-time.After(r.retryAfter):
-			Abandon().Do(ctx, orig, message)
-		}
-	}()
+	select {
+	// TODO this can go past lock duration pretty easily
+	// Ideally we'd also timeout at dequeue time + lockdurtaion - 1 second but don't have access to dequeue time
+	// Maybe we can use context.WithTimeout when we recieve the message?
+	case <-ctx.Done():
+		span.Logger().Error(errors.Wrap(ctx.Err(), "Retry context expired"))
+	case <-time.After(r.retryAfter):
+		Abandon().Do(ctx, orig, message)
```

Comment:
should we stop abandoning now that we have peek lock?

Reply:
mmh, I don't think we should.
you retry after a given time.
you can say after 15 minutes... even if your lock is only 2 min, that will work thanks to the auto-renewal.

plus, it makes the handler cleaner.
I can now remove the unused orginalHandler passed to every handler :)